zulipterminal/api_types.py

```diff
@@ -197,37 +197,26 @@ class Message(TypedDict, total=False):
 
 
 ###############################################################################
-# In "subscriptions" response from:
+# In "subscriptions", "unsubscribed", and "never_subscribed" responses from:
 #   https://zulip.com/api/register-queue
 # Also directly from:
 #   https://zulip.com/api/get-events#subscription-add
 #   https://zulip.com/api/get-subscriptions (unused)
 
 
-class Subscription(TypedDict):
+class Stream(TypedDict):
     stream_id: int
     name: str
     description: str
     rendered_description: str
-    date_created: int  # NOTE: new in Zulip 4.0 / ZFL 30
+    date_created: Optional[int]  # NOTE: new in Zulip 4.0 / ZFL 30
```

Comment:
Yes, the part where we add the `date_created` key and set it to None fails the typecheck. Since we get the data from the server initially, mypy doesn't catch that `date_created` is not present. We could account for this by using `NotRequired[Optional[int]]` instead, but that would purely be for documentation since it passes the typecheck with just `Optional[int]`. I'll also add a comment.